src/python/pants/backend/python/providers/pyenv/rules.py

```diff
@@ -0,0 +1,183 @@
+from pants.engine.platform import Platform
+from pants.engine.process import Process, ProcessCacheScope, ProcessResult
+from pants.engine.rules import collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.option.option_types import StrListOption
+from pants.util.frozendict import FrozenDict
+from pants.util.meta import classproperty
+
+
+class PyenvSubsystem(TemplatedExternalTool):
+    options_scope = "pyenv"
+    name = "pyenv"
+    help = "pyenv (https://github.com/pyenv/pyenv)."
+
+    default_version = "2.3.13"
+    default_url_template = "https://github.com/pyenv/pyenv/archive/refs/tags/v{version}.tar.gz"
+
+    python_configure_opts = StrListOption(
+        help="Flags to use when configuring CPython.",
+        advanced=True,
+    )
+
+    @classproperty
+    def default_known_versions(cls):
+        return [
+            "|".join(
+                (
+                    cls.default_version,
+                    plat,
+                    "9105de5e5cf8dc0eca2a520ed04493d183128d46a2cfb402d4cc271af1bf144b",
+                    "749323",
+                )
+            )
+            for plat in ["macos_arm64", "macos_x86_64", "linux_x86_64", "linux_arm64"]
+        ]
+
+    def generate_exe(self, plat: Platform) -> str:
+        """Returns the path to the tool executable.
+
+        If the downloaded artifact is the executable itself, you can leave this unimplemented.
+
+        If the downloaded artifact is an archive, this should be overridden to provide a
+        relative path in the downloaded archive, e.g. `./bin/protoc`.
+        """
+        return f"./pyenv-{self.version}/bin/pyenv"
+
+
+class PythonToolchainRequest(PythonProvider):
+    pass
+
+
+PYENV_NAMED_CACHE = ".pyenv"
+PYENV_APPEND_ONLY_CACHES = FrozenDict({"pyenv": PYENV_NAMED_CACHE})
+
+
+@rule
+async def get_python(
+    request: PythonToolchainRequest,
+    python_setup: PythonSetup,
+    platform: Platform,
+    pyenv_subsystem: PyenvSubsystem,
+) -> PythonExecutable:
+    env_vars, pyenv = await MultiGet(
+        Get(EnvironmentVars, EnvironmentVarsRequest(["PATH", "LDFLAGS"])),
+        Get(DownloadedExternalTool, ExternalToolRequest, pyenv_subsystem.get_request(platform)),
+    )
+
+    python_to_use = request.interpreter_constraints.minimum_python_version(
+        python_setup.interpreter_versions_universe
+    )
+    if python_to_use is None:
+        raise ValueError(
+            f"Couldn't determine a compatible Interpreter Constraint from {python_setup.interpreter_versions_universe}"
+        )
+
+    which_python_result = await Get(
+        ProcessResult,
+        Process(
+            [pyenv.exe, "latest", "--known", python_to_use],
+            input_digest=pyenv.digest,
+            description=f"Choose specific version for Python {python_to_use}",
+            env={"PATH": env_vars.get("PATH", "")},
+            # Caching the result is OK, since if the user really needs a different patch,
```

Comment:
I concur.  And this will only matter when new Python versions are released, which is... not often.